frontend/wallet/src/render/components/AddWalletModal.re

```diff
@@ -44,7 +44,10 @@ let make = (~modalState, ~setModalState) => {
         <Button
           label="Create"
           style=Button.Green
-          onClick={_ => setModalState(_ => None)}
+          onClick={_ => {
+            onSubmit(Option.getExn(modalState));
```

Comment:
It sucks that we need to do this instead of pattern matching, but as discussed in person, given that the Modal has an annoying isOpen, it's okay...